pkg/util/config/config.go

```diff
@@ -67,21 +68,25 @@ type Config struct {
 	Resources
 	Auth
 	Scalyr
-	WatchedNamespace          string `name:"watched_namespace"` // special values: "*" means 'watch all namespaces', the empty string "" means 'watch a namespace where operator is deployed to'
-	EtcdHost                  string `name:"etcd_host" default:"etcd-client.default.svc.cluster.local:2379"`
-	DockerImage               string `name:"docker_image" default:"registry.opensource.zalan.do/acid/spiloprivate-9.6:1.2-p4"`
-	ServiceAccountName        string `name:"service_account_name" default:"operator"`
-	DbHostedZone              string `name:"db_hosted_zone" default:"db.example.com"`
-	EtcdScope                 string `name:"etcd_scope" default:"service"`
-	WALES3Bucket              string `name:"wal_s3_bucket"`
-	KubeIAMRole               string `name:"kube_iam_role"`
-	DebugLogging              bool   `name:"debug_logging" default:"true"`
-	EnableDBAccess            bool   `name:"enable_database_access" default:"true"`
-	EnableTeamsAPI            bool   `name:"enable_teams_api" default:"true"`
-	EnableTeamSuperuser       bool   `name:"enable_team_superuser" default:"false"`
-	TeamAdminRole             string `name:"team_admin_role" default:"admin"`
-	EnableMasterLoadBalancer  bool   `name:"enable_master_load_balancer" default:"true"`
-	EnableReplicaLoadBalancer bool   `name:"enable_replica_load_balancer" default:"false"`
+	PodServiceAccount v1.ServiceAccount // has to be struct value, not a pointer
```

Comment:
Why should it be part for the Config structure; I'd rather move it to the controller itself; also, why cannot it be a pointer?

Reply:
This service account is accessed during cluster creation through the copy of the operator `Config` struct present in the every `Cluster` struct.

> Why should it be part for the Config structure; I'd rather move it to the controller itself

that is also an option, by what then would be a correct way to reference this `Controller` struct from within cluster `Create()` ?

> why cannot it be a pointer?

just to prevent potential race condition in the case the account needs to be deployed to two different namespaces simultaneously.

Reply:
> that is also an option, by what then would be a correct way to reference this Controller struct from within cluster Create() ?

We can also copy it when the cluster is created from the controller into the cluster. The point is that it doesn't belong to the operator config, I'd rather have a separate field in the Cluster structure for it.

> To prevent potential race condition in the case the account needs to be deployed to two different namespaces simultaneously. If it were a pointer, then two copies of the operator Config might inadvertently re-use the same "namespace" field in the same service account struct.

>To prevent potential race condition in the case the account needs to be deployed to two different namespaces simultaneously. If it were a pointer, then two copies of the operator Config might inadvertently re-use the same "namespace" field in the same service account struct.

Well, Create at the end of the day is just a REST API call that shouldn't modify the definition passed to it. I think you can remove the namespace setting from the service description altogether and rely on the Services(Namespace) to create the service account in the right namespace.